asterix/cat_002.ksy

```diff
@@ -0,0 +1,166 @@
+meta:
+  id: cat_002
+  license: GPL-3.0-only
+  endian: be
+  imports:
+      - field_spec
+      - explicit
+      - cat_002_catalog
+  ks-version: 0.9
+
+doc: |
+  Implementantion of ASTERIX format.
+
+  From Eurocontrol Asterix web page at
+  https://www.eurocontrol.int/asterix
+
+  Category 002 definition file
+  Radar Data Exchange Part 2b
+  Transmission of Monoradar Service Messages
+  SUR.ET1.ST05.2000-STD-02b-01
+  Edition : 1.0
+
+seq:
+  - id: fspec
+    type: field_spec
+
+  - id: data
+    type: uap_type(fspec, [10,0,20,30,41,50,60,70,100,90,80].as<u2[]>.as<str>)
+
+types:
+  uap_type:
+    params:
+
+      - id: fspec
+        type: field_spec
+
+      - id: uap
+        type: str
+
+    seq:
+      - id: items
+        type:
+          switch-on: fspec.octects[_index/7].bits[_index % 7]
+          cases:
+            true: cat_002_catalog(uap_list[ _index ])
+        repeat: expr
+        repeat-expr: loop_range
+
+
+      - id: sp #,"E","1","Special Purpose Field","N.A."),
+        type: explicit(true)
+        if: fspec.size >= 13 and fspec.octects[1].bits[5]
+
+      - id: irfs #,"E","1","Reserved Expansion Field","N.A."),
+        type: explicit(true)
+        if: fspec.size >= 14 and fspec.octects[1].bits[6]
+
+
+    instances:
+
+      uap_list:
+        value: uap.as<u2[]>
+
+      loop_range:
+        value: 'fspec.size > uap_list.size ? uap_list.size : fspec.size'
+
+instances:
+
+  category:
+    value: '"002"'
+
+  version:
+    value: '"1.00"'
+
+  i010:
```

Comment:
I guess the names should be more meaningful

Reply:
i010, is ITEM 010, from Catalog, as named by the Corresponding Category definition. i010 is the only name, there is no other. Then once you access the item, you could decode various possible values encoded in it. for example Item 10, is a 2 octect field, that has encoded two values SIC and SAC, this has to be decoded by calling program or in instance value. But the important thing here is that if you need to decode a category that is not in the repository you could define new one in just 10 minutes, and possible contribute it.